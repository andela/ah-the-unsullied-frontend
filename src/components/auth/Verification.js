import React, { Component } from 'react';
import { connect } from 'react-redux';
import { verifyUser } from '../../actions/verification';
import { Link } from 'react-router-dom';

class Verification extends Component {
  state = {
    errors: {}
  };

  componentDidMount() {
    const { verifyUser: verifyAction } = this.props;
    const token = this.props.match.params.token;
    const pk = this.props.match.params.pk;

    verifyAction({ pk, token });
  }

  componentWillReceiveProps(nextprops) {}

  render() {
    const response = this.props.verify;
    return (
      <div className="container ">
        <div className="row valign-wrapper center-align">
          <div className="col s6 offset-s3 valign">
            <div className="card grey darken-1">
              <div className="card-content white-text">
                <span className="card-title">Verification</span>
                {response.errors ? (
                  <p style={{ color: 'red' }}>{response.errors.data}</p>
                ) : null}
                {response.data ? (
                  <p style={{ color: 'white' }}>{response.data}</p>
                ) : null}
              </div>
              <div className="card-action">
                <p>
                  Proceed to login into your account <Link to="/">Login</Link>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    verify: state.verifyEmail
  };
};
export default connect(
  mapStateToProps,
  { verifyUser }
)(Verification);
