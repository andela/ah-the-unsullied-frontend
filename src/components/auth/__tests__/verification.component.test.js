import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Verification from '../Verification';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    verifyUser: jest.fn()
  };
  const wrapper = shallow(<Verification {...props} />);

  return {
    props,
    wrapper
  };
}

describe('Verification elements tests', () => {
  it('renders a the verification elements', () => {
    const { wrapper } = setup();
    expect(wrapper.find('card')).toBeDefined();
    expect(wrapper.find('button')).toBeDefined();
  });
});
