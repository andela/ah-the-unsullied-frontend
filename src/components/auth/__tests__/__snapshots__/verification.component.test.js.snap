// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verification should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Verification
    match={
      Object {
        "params": Object {
          "token": "fjdsafjkdajkfjkjkvdjskaahjsdhsjfhjkhgdsjfdfhsdjkhajkfhgrlk",
        },
      }
    }
    verify="sa@gmail.com"
    verifyUser={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "pk": undefined,
              "token": "fjdsafjkdajkfjkjkvdjskaahjsdhsjfhjkhgdsjfdfhsdjkhajkfhgrlk",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row valign-wrapper center-align"
      >
        <div
          className="col s6 offset-s3 valign"
        >
          <div
            className="card grey darken-1"
          >
            <div
              className="card-content white-text"
            >
              <span
                className="card-title"
              >
                Verification
              </span>
            </div>
            <div
              className="card-action"
            >
              <p>
                Proceed to login into your account 
                <Link
                  replace={false}
                  to="/"
                >
                  Login
                </Link>
              </p>
            </div>
          </div>
        </div>
      </div>,
      "className": "container ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col s6 offset-s3 valign"
        >
          <div
            className="card grey darken-1"
          >
            <div
              className="card-content white-text"
            >
              <span
                className="card-title"
              >
                Verification
              </span>
            </div>
            <div
              className="card-action"
            >
              <p>
                Proceed to login into your account 
                <Link
                  replace={false}
                  to="/"
                >
                  Login
                </Link>
              </p>
            </div>
          </div>
        </div>,
        "className": "row valign-wrapper center-align",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card grey darken-1"
          >
            <div
              className="card-content white-text"
            >
              <span
                className="card-title"
              >
                Verification
              </span>
            </div>
            <div
              className="card-action"
            >
              <p>
                Proceed to login into your account 
                <Link
                  replace={false}
                  to="/"
                >
                  Login
                </Link>
              </p>
            </div>
          </div>,
          "className": "col s6 offset-s3 valign",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-content white-text"
              >
                <span
                  className="card-title"
                >
                  Verification
                </span>
              </div>,
              <div
                className="card-action"
              >
                <p>
                  Proceed to login into your account 
                  <Link
                    replace={false}
                    to="/"
                  >
                    Login
                  </Link>
                </p>
              </div>,
            ],
            "className": "card grey darken-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="card-title"
                  >
                    Verification
                  </span>,
                  null,
                  null,
                ],
                "className": "card-content white-text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Verification",
                    "className": "card-title",
                  },
                  "ref": null,
                  "rendered": "Verification",
                  "type": "span",
                },
                null,
                null,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Proceed to login into your account 
                  <Link
                    replace={false}
                    to="/"
                  >
                    Login
                  </Link>
                </p>,
                "className": "card-action",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Proceed to login into your account ",
                    <Link
                      replace={false}
                      to="/"
                    >
                      Login
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Proceed to login into your account ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Login",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row valign-wrapper center-align"
        >
          <div
            className="col s6 offset-s3 valign"
          >
            <div
              className="card grey darken-1"
            >
              <div
                className="card-content white-text"
              >
                <span
                  className="card-title"
                >
                  Verification
                </span>
              </div>
              <div
                className="card-action"
              >
                <p>
                  Proceed to login into your account 
                  <Link
                    replace={false}
                    to="/"
                  >
                    Login
                  </Link>
                </p>
              </div>
            </div>
          </div>
        </div>,
        "className": "container ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col s6 offset-s3 valign"
          >
            <div
              className="card grey darken-1"
            >
              <div
                className="card-content white-text"
              >
                <span
                  className="card-title"
                >
                  Verification
                </span>
              </div>
              <div
                className="card-action"
              >
                <p>
                  Proceed to login into your account 
                  <Link
                    replace={false}
                    to="/"
                  >
                    Login
                  </Link>
                </p>
              </div>
            </div>
          </div>,
          "className": "row valign-wrapper center-align",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card grey darken-1"
            >
              <div
                className="card-content white-text"
              >
                <span
                  className="card-title"
                >
                  Verification
                </span>
              </div>
              <div
                className="card-action"
              >
                <p>
                  Proceed to login into your account 
                  <Link
                    replace={false}
                    to="/"
                  >
                    Login
                  </Link>
                </p>
              </div>
            </div>,
            "className": "col s6 offset-s3 valign",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-content white-text"
                >
                  <span
                    className="card-title"
                  >
                    Verification
                  </span>
                </div>,
                <div
                  className="card-action"
                >
                  <p>
                    Proceed to login into your account 
                    <Link
                      replace={false}
                      to="/"
                    >
                      Login
                    </Link>
                  </p>
                </div>,
              ],
              "className": "card grey darken-1",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="card-title"
                    >
                      Verification
                    </span>,
                    null,
                    null,
                  ],
                  "className": "card-content white-text",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Verification",
                      "className": "card-title",
                    },
                    "ref": null,
                    "rendered": "Verification",
                    "type": "span",
                  },
                  null,
                  null,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    Proceed to login into your account 
                    <Link
                      replace={false}
                      to="/"
                    >
                      Login
                    </Link>
                  </p>,
                  "className": "card-action",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Proceed to login into your account ",
                      <Link
                        replace={false}
                        to="/"
                      >
                        Login
                      </Link>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Proceed to login into your account ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Login",
                        "replace": false,
                        "to": "/",
                      },
                      "ref": null,
                      "rendered": "Login",
                      "type": [Function],
                    },
                  ],
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
